src/main/resources/META-INF/rewrite/spring-data-bom-2024.1.yml

```diff
@@ -0,0 +1,25 @@
+#
+# Copyright 2025 the original author or authors.
+# <p>
+# Licensed under the Moderne Source Available License (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://docs.moderne.io/licensing/moderne-source-available-license
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+########################################################################################################################
+# Spring Data 2024.1 (https://spring.io/projects/spring-data)
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.spring.data.UpgradeSpringData_2024_1
+displayName: Migrate to Spring Data 2024.1
+description: Migrate applications to the latest Spring Data 2024.1 release.
+recipeList:
+  - org.openrewrite.java.spring.data.UpgradeSpringDataJPA_3_4
```

Comment:
I think I'd prefer to see `UpgradeSpringDataJPA_3_4` included with the Spring Boot upgrade recipe, as opposed to this Spring Data upgrade that folks would need to run separately.